client/components/bloganuary-header/index.tsx

```diff
@@ -0,0 +1,90 @@
+import { Button } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useRef } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import BloganuaryIcon from 'calypso/components/blogging-prompt-card/bloganuary-icon';
+import isBloganuary from 'calypso/data/blogging-prompt/is-bloganuary';
+import { Tag } from 'calypso/reader/list-manage/types';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { successNotice } from 'calypso/state/notices/actions';
+import { requestFollowTag } from 'calypso/state/reader/tags/items/actions';
+import { getReaderFollowedTags } from 'calypso/state/reader/tags/selectors';
+import { toggleReaderSidebarTags } from 'calypso/state/reader-ui/sidebar/actions';
+import { isTagsOpen } from 'calypso/state/reader-ui/sidebar/selectors';
+import './style.scss';
+
+const containsBloganuary = ( followedTags: Tag[] | undefined ): boolean | undefined => {
+	return followedTags?.some( ( tag: Tag ) => tag.slug === 'bloganuary' );
+};
+
+const BloganuaryHeader = () => {
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+	const followedTags = useSelector( getReaderFollowedTags ) as Tag[];
+	const prevIsFollowingBloganuary = useRef( containsBloganuary( followedTags ) );
+	const isFollowingBloganuary = containsBloganuary( followedTags );
+	const isTagsSidebarOpen = useSelector( isTagsOpen );
+
+	useEffect( () => {
+		dispatch( recordTracksEvent( 'calypso_reader_bloganuary_banner_view' ) );
+	}, [ dispatch ] );
```

Comment:
It feels weird using `dispatch` as a dependency in these cases (although I get the linters highlight it when its used in the hook). Ideally, dispatch should never change, but if for some reason its reference in memory changes this would result in sending the `*view` tracking event extra times and messing with the counts past what an empty dependency array would. 🤔 

I think we have other methods of `recordTracksEvent` that don't need to be used with dispatch, the dispatch version builds on them, but im unsure if its actually buying us anything valuable in this case.  I asked about this in slack a few weeks back p1697648257681619-slack-C03N25JPCE4

Reply:
good catch! ahhh yes, I thought I'd seen recordTracksEvent work without dispatch! it was a different function with the same name 😆